src/main/java/com/cloudbees/hudson/plugins/folder/views/DefaultFolderViewHolder.java

```diff
@@ -122,4 +132,22 @@ public void setTabBar(ViewsTabBar tabBar) {
         this.tabBar = tabBar;
     }
 
+    private Object readResolve() throws ObjectStreamException {
+        if (primaryView != null) {
+            // TODO replace reflection with direct access once baseline core has JENKINS-38606 fix merged
+            try {
+                Method migrateLegacyPrimaryAllViewLocalizedName = AllView.class
+                        .getMethod("migrateLegacyPrimaryAllViewLocalizedName", List.class, String.class);
+                primaryView =
+                        (String) migrateLegacyPrimaryAllViewLocalizedName.invoke(null, views, primaryView);
+            } catch (NoSuchMethodException e) {
+                // ignore, Jenkins core does not have JENKINS-38606 fix merged
+            } catch (IllegalAccessException e) {
+                // ignore, Jenkins core does not have JENKINS-38606 fix merged
+            } catch (InvocationTargetException e) {
+                // ignore, Jenkins core does not have JENKINS-38606 fix merged
```

Comment:
ditto

Reply:
And yet this _is thrown, routinely, and then swallowed just as I had warned: [JENKINS-47416](https://issues.jenkins-ci.org/browse/JENKINS-47416)